File: nodejs/sampleServer.js
var express = require('express');
var bodyParser = require('body-parser');
var fs = require('fs');
var crypto = require('crypto');

var app = express();
var port = process.env.PORT || 8080;
app.use(bodyParser.json());

var lines = null

fs.readFile('../text.txt', 'utf8', function(err, data) {
    if (err) throw err;
    lines = data.split(/\r?\n/);
});

app.get('/nodejs/write', function(req, res) {
	var line_num = parseInt(req.param('l'));
	if (isNaN(line_num)) {
		res.status(400).send('invalid number')
	}
	else if( line_num < 1 || line_num > 100) {
		res.status(400).send('out of range')
	} else {
		res.send(lines[line_num]);
	}
});

app.post('/nodejs/sha256', function(req, res) {
	var n1 = req.body.n1;
	var n2 = req.body.n2;
	if ((typeof n1) != "number" || (typeof n2) != "number") {
		res.status(400).send('invalid numbers')
	} else {
		var hash = crypto.createHash('sha256').update((n1 + n2).toString()).digest('base64');
		res.type('json');
		res.end(JSON.stringify({result: hash}));
	}
});


app.listen(port);
console.log('Server started! At http://localhost:' + port);

File: go/sampleServer.go
// +build ignore

package main

import (
	"log"
    "fmt"
    "net/http"
	"crypto/sha256"
	"strconv"
	"strings"
	"encoding/json"
	"io/ioutil"
)

var lines []string

type nums struct {
	N1 int `json:"n1"`
	N2 int `json:"n2"`
}

type result struct { Result string `json:"result"` }

func handlerSha(w http.ResponseWriter, r *http.Request) {
	w.Header().Set("Content-Type", "application/json")

	decoder := json.NewDecoder(r.Body)
	
	var data nums
	err := decoder.Decode(&data)
	if err != nil {
		w.WriteHeader(http.StatusBadRequest)
		fmt.Fprintf(w, "invalid numbers")
	} else {
		n1 := data.N1
		n2 := data.N2
		var sum = hashIt(n1, n2)
		fmt.Fprintf(w, "%s", sum)
	}
}

func hashIt(n1 int, n2 int) []byte {
	var n = n1 + n2
	s := strconv.Itoa(n)	
    shaa := sha256.Sum256([]byte(s))
	sha := fmt.Sprintf("%x", shaa)
	resObj := &result{ Result: sha }
	res, _ := json.Marshal(resObj)
	return res
}

func handlerWrite(w http.ResponseWriter, r *http.Request) {
	query := r.URL.Query()
    lineNumStr := query.Get("l")
	lineNum, err := strconv.Atoi(lineNumStr)
	if err != nil {
		w.WriteHeader(http.StatusBadRequest)
		fmt.Fprintf(w, "invalid number")
	} else if (lineNum < 1 || lineNum > 100) {
		w.WriteHeader(http.StatusBadRequest)
		fmt.Fprintf(w, "out of range")
	} else {
		result := lines[lineNum]
		fmt.Fprintf(w, "%s", result)
	}
}

func main() {
	content, _ := ioutil.ReadFile("/root/server/text.txt")
	text := string(content)
	lines = strings.Split(text, "\n")
	fmt.Println("start")
    http.HandleFunc("/go/sha256", handlerSha)
	http.HandleFunc("/go/write", handlerWrite)
    log.Fatal(http.ListenAndServe(":8082", nil))
}


